wordpress-mobile/WordPress-Android
# Issue/8497 copy site creation flow

Fixes #8497 

Goal of this ticket is to create a copy of the current Site Creation flow so we don't have to worry about breaking the old flow as we plan to run A/B tests.

- I copy-pasted all the related classes to a new package
- I introduced a new feature flag -> we'll replace the build time feature flag with a flag from Remote Firebase config as soon as we have support for it


To test:
1. Verify none of the legacy classes has been modified
2. Verify the new classes don't reference legacy classes (org.wordpress.android.ui.accounts.signup)
3. Verify the feature flag is correct and the app uses the legacy SiteCreation flow

Basically assume that we'll merge this PR to the develop within a week or so and it shouldn't have any effect whatsoever.

Note:
The change set is huge, but I just copy-pasted all the related classes. I believe you can review just e1d56d6 and 2395203 + `To test` steps.
